File: TP1/Java/Soma_Digitos/Somad.java
import java.util.Scanner;

public class Somad{

	public static int soma(String num, int indice){
		if (indice < 0) {
            return 0;
        }
        return (num.charAt(indice) - '0') + soma(num, indice - 1);
	}

	public static void main(String[] args){
		Scanner entrada = new Scanner(System.in);
		
		do{
			String num = entrada.nextLine();
			if(num.equals("FIM")){
				break;
			}

			int tam = num.length();
			int result = soma(num, tam - 1);
			System.out.println(""+result);

		}while(true);

		entrada.close();
	}
}

File: TP1/C/Soma_Digitos/Somad.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

int soma(char num[], int indice) {
    if (indice < 0) {
        return 0;
    }

    return (num[indice] - '0') + soma(num, indice - 1);
}

int main() {
    char numeros[100];

    do {
        fgets(numeros, sizeof(numeros), stdin);
        numeros[strcspn(numeros, "\n")] = '\0'; 

        if (strcmp(numeros, "FIM") == 0) {
            break;
        }

        int tam = strlen(numeros);
        int result = soma(numeros, tam - 1);
        printf("%d\n", result);

    } while (true);

    return 0;
}
